src/pages/fi-opening-balances/forms/FiOpeningBalancesForm.js

```diff
@@ -0,0 +1,222 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import { useForm } from 'src/hooks/form'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+
+export default function FiOpeningBalancesForms({ labels, maxAccess, recordId }) {
+  const [isLoading, setIsLoading] = useState(false)
+  const [editMode, setEditMode] = useState(!!recordId)
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: FinancialRepository.FiOpeningBalance.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: null,
+      accountId: null,
+      fiscalYear: null,
+      amount: null,
+      baseAmount: null,
+      currencyId: null,
+      plantId: null,
+      plantName: '',
+      plantRef:''
```

Comment:
do you need plantName and plantRef in the formik ? it's a combo and it will set the plantName from the plantId

Reply:
yes if we just save the id the data is not saved correctly